### PR TITLE
Update 2.x with changes from 2.2.x and 2.3.x

### PR DESCRIPTION
**Pull request by m-vdb**
## Description

The goal of this PR is to show how changes made in a minor version would be documented in a "Docs for major only" setup. More information in [this document](https://www.notion.so/rasa/Doc-versions-for-Majors-only-caaf78eadff34d7d8b49eb2284ccc8f9).

## How to use this PR
- you can run do `cd docs && yarn && yarn develop` to run the docs locally
- skip `changelog.mdx`
- skip `sources/` and `reference/` folders
- look at different kind of changes:
  - changes introduced in the docs because there were changes in the codebase (new feature, changes in behaviour, deprecations)
  - changes in the docs to improve it, independent of codebase change
 
<details><summary>💡 Pro-tip, filter files in the PR on ".mdx" only</summary>
Screenshot 2021-08-05 at 13 48 11</details>

## Next steps
- [ ] get reviews
- [ ] remove the 2.2.x version of the docs in this branch
- [ ] remove the 2.3.x version of the docs in this branch
- [ ] merge this branch into `documentation-full`
- [ ] repeat the process for all minors until 2.8

### CODE REVIEW

**Comment by m-vdb** [comment]
@desmarchris I went ahead and merged 2.2.x and 2.3.x in this PR. Can you take a look when you have the chance? Changes were easier to make this time, so I did 2 versions in a row

File: docs/versioned_docs/version-2.x/nlu-only-server.mdx [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+by adding the connection details to the dialogue management server's endpoint configuration file:
+
+```yaml title="endpoints.yml"
+nlu:
+    url: "http://<your nlu host>:<your nlu port>"
+    token: <token>  # [optional]
+    token_name: <name of the token> # [optional] (default: token)
+```
+
+The `token` and `token_name` refer to optional [authentication paramenters](./http-api.mdx#token-based-auth).
```
Comment (desmarchris) [inline docs/versioned_docs/version-2.x/nlu-only-server.mdx:22]:
```suggestion
The `token` and `token_name` refer to optional [authentication parameters](./http-api.mdx#token-based-auth).
```

**Review by desmarchris — CHANGES_REQUESTED** [review]

**Comment by desmarchris** [comment]
looks good!

**Comment by m-vdb** [comment]
@desmarchris probably a good idea to approve the changes if the requested changes are minor. That way I can incorporate the changes and then merge the PR without overriding your change request (here I assume that everything is good apart from the minor change). Not a big deal :)

**Pull request merged**